torch/_dynamo/variables/misc.py

```diff
@@ -78,7 +90,35 @@ def _resolved_getattr_and_source(self, tx, name):
                     break
 
         # TODO(jansel): there is a small chance this could trigger user code, prevent that
-        return getattr(super(search_type, type_to_use), name), source
+        # It is important to call with objvar here to ensure that when we get methods,
+        # they are bound appropriately
+        # Note that what gets bound here for methods depend on how the name method is defined
+        # and does not depend on objvar being self or cls in the current function.
+        objvar_python = (
+            self.objvar.module
+            if isinstance(self.objvar, NNModuleVariable)
+            else self.objvar.value
+        )
+        bound_method = getattr(super(search_type, objvar_python), name)
```

Comment:
This has a few issues:
1) It doesn't handle all VariableTracker types.  You are reading `objar.value` which might not exist.
2) `getattr()` can invoke user code via `__getattribute__` `__getattr__`.  We could use `inspect.getattr_static`, or you could check `obj.__dict__`.

Can we do this without needing to read the python value?  Perhaps we can just have handling for `UserDefinedClassVariable` since I think that is the isssue being fixed here?

Reply:
Maybe something like:
```
if isinstance(self.objvar, UserDefinedClassVariable):
   search_type = self.objvar.as_python_constant()
```

Reply:
I don't think I understand this subtlety enough to properly fix it.
And the case of a method calling a classmethod is not the most common so removing it from this PR.

I left the extra comments I added. let me know if you prefer I remove them.